### ISSUE TITLE
Parallelize the sampling

### ISSUE THREAD

**Issue by NicoRenaud** [NLESC-JCER/DeepQMC#1]
The MC sampling is now done on 1 proc only. This is trivial to parallelize but we could use different options : mpi4py, multiprocess, ....

**Comment by NicoRenaud** [comment]
The `multiprocessing` route is more difficult than expected. See the `mp_sampling` branch

**Comment by NicoRenaud** [comment]
There is a native `distriubted` module in pytorch : https://pytorch.org/tutorials/intermediate/dist_tuto.html

**Comment by NicoRenaud** [comment]
Instead of parallelizing the sampler, It might be best to use `DistributedDataParallel`to directly parallelize the training (https://pytorch.org/docs/stable/nn.html#torch.nn.parallel.DistributedDataParallel)

We could also split the number of walker in the training function and average the gradients as done there https://pytorch.org/tutorials/intermediate/dist_tuto.html

**Comment by NicoRenaud** [comment]
The average gradients method seems to be the best for us. However a MPI backend would simplify stuff
https://medium.com/intel-student-ambassadors/distributed-training-of-deep-learning-models-with-pytorch-1123fa538848

**Comment by NicoRenaud** [comment]
The DistributedDataParallel of pytoch is too hard to use. I'll switch to Hovorod instead
